api/v1alpha1/timeout_types.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+// Timeout defines configuration for timeouts related to connections.
+type Timeout struct {
+	// Timeout settings for TCP.
+	//
+	// +optional
+	TCP *TCPTimeout `json:"tcp,omitempty"`
+
+	// // Timeout settings for HTTP.
+	//
+	// +optional
+	HTTP *HTTPTimeout `json:"http,omitempty"`
+}
+
+type TCPTimeout struct {
+	// The timeout for new network connection establishment, including TCPTimeout and TLS handshakes.
+	// Default: 10 seconds.
+	//
+	// +optional
+	ConnectTimeout *gwapiv1.Duration `json:"connectTimeout,omitempty"`
+}
+
+type HTTPTimeout struct {
+	// The idle timeout for persistent HTTPTimeout connections. Idle time is defined as a period in which there are no active requests in the connection.
+	// Default: 1 hour.
+	//
+	// +optional
+	ConnectionIdleTimeout *gwapiv1.Duration `json:"connectionIdleTimeout,omitempty"`
```

Comment:
is the `connection` prefix needed here ?

Reply:
Envoy also has `stream` timeouts, e.g. [stream idle timeout](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-stream-idle-timeout). So, to distnguish between the two, I added the connection prefix.

Reply:
sgtm, thanks